### PR TITLE
refactor: middleware/redirects.ts

### PR DESCRIPTION
**Pull request by pbkompasz**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring

## Context

- [X] Part of #6782
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [X] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt&usdamount=50&donation=true)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 1c31624</samp>

Refactored the redirect logic in `middleware/redirects.ts` to use an array of objects and regular expressions. Improved readability, maintainability, and scalability of the code.

### <samp>🤖 Generated by Copilot at 1c31624</samp>

> _`redirects` change_
> _array of objects, regex_
> _console logs in spring_

### CODE REVIEW

File: middleware/redirects.ts [hunk 1]
```diff
@@ -1,27 +1,49 @@
...
 
-  if (route.path.startsWith(`/${urlPrefix.value}`)) {
-    if (route.path.endsWith('collections')) {
-      return redirect(`/${urlPrefix.value}/explore/collectibles`)
-    }
-    if (route.path.endsWith('gallery')) {
-      return redirect(`/${urlPrefix.value}/explore/items`)
-    }
-  }
+  console.log(route.path)
```
Comment (stephenjason89) [inline middleware/redirects.ts:6]: remove this.
```suggestion
```

File: middleware/redirects.ts [hunk 2]
```diff
@@ -1,27 +1,49 @@
...
+        window.location.href.replace('/transfer', '/ksm/transfer'),
+    },
+  ]
 
-  if (route.path.includes('/rmrk2/')) {
-    return redirect(window.location.href.replace('/rmrk2/', '/ksm/'))
+  for (const path of paths) {
+    if (path.cond(route.path)) {
+      redirectValue = path.replaceValue()
+      console.log(route.path, redirectValue)
```
Comment (stephenjason89) [inline middleware/redirects.ts:41]: remove this as well.
```suggestion
```

File: middleware/redirects.ts [hunk 3]
```diff
@@ -1,27 +1,49 @@
...
-  }
+  const paths = [
+    {
+      cond: (val) => val === '/drops',
+      replaceValue: () => '/ahk/drops',
+    },
+    {
+      replaceValue: () => `/${urlPrefix.value}/explore/collectibles`,
+      cond: (val) =>
+        new RegExp(`^\\/${urlPrefix.value}\\/.*\\/collections$`).test(val),
```
Comment (stephenjason89) [inline middleware/redirects.ts:16]: startsWith and endsWith is more readable imo.

File: middleware/redirects.ts [hunk 4]
```diff
@@ -1,27 +1,49 @@
...
-  }
+  console.log(route.path)
 
-  if (route.path.includes('/stmn/')) {
-    return redirect(window.location.href.replace('/stmn/', '/ahk/'))
-  }
+  const paths = [
+    {
+      cond: (val) => val === '/drops',
+      replaceValue: () => '/ahk/drops',
```
Comment (stephenjason89) [inline middleware/redirects.ts:11]: does replaceValue need to be a function? 
I believe it should work even if it is not. 
maybe 
```
{
      cond: (val) => val === '/drops',
      url: '/ahk/drops',
}
```

Comment (pbkompasz) [reply middleware/redirects.ts:10]: I can change to simple string and check if it's a callable function.

Comment (stephenjason89) [reply middleware/redirects.ts]: @pbkompasz  that is not what i meant. Sorry let me explain, all replaceValue field doesn't need to be a function i believe. Is there a reason why these are functions?

**Comment by prury** [comment]
So, if i am on BSX and type /transfer on address bar it will always redirect me to ksm/transfer? 
@vikiival is this behavior correct? 
shouldn't it redirect me to current prefix/transfer?

**Comment by stephenjason89** [comment]
> 

Yes @prury, upon checking that's also the old behavior and has just been rewritten. 
I do not know if it is the correct behavior though. 
@vikiival please confirm. Thank you so much.

File: middleware/redirects.ts [hunk 5]
```diff
@@ -1,27 +1,49 @@
...
+      break
+    }
   }
 
-  if (route.path.includes('/rmrk2/')) {
-    return redirect(window.location.href.replace('/rmrk2/', '/ksm/'))
-  }
-  if (route.path.startsWith('/transfer')) {
-    return redirect(window.location.href.replace('/transfer', '/ksm/transfer'))
+  if (redirect) {
```
Comment (Jarsen136) [inline middleware/redirects.ts:46]: This condition will always be true.

```suggestion
  if (redirectValue) {
```

**Review by Jarsen136 — CHANGES_REQUESTED** [review]

**Comment by vikiival** [comment]
> shouldn't it redirect me to current prefix/transfer?

I think it was for legacy purposes but my bet is @yangwao does not need this.
Therefore it should redirect prefix/transfer

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.56 USD/DOT ~ 6.579 $DOT
🧗 [16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt](https://polkadot.subscan.io/account/16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt)
🔗 [0x0da2507e8c7bddf805ea58afd8090a231e19a08f9e07fb4c0d62154c26275ec2](https://polkadot.subscan.io/extrinsic/0x0da2507e8c7bddf805ea58afd8090a231e19a08f9e07fb4c0d62154c26275ec2)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**